test/crash_course_test.exs

```diff
@@ -1,8 +1,58 @@
-defmodule CrashCourseTest do
+defmodule CrashCourseAgentTest do
   use ExUnit.Case
-  doctest CrashCourse
+  doctest CrashCourseAgent
 
-  test "greets the world" do
-    assert CrashCourse.hello() == :world
+  use ExUnit.Case
+
+  test 'launches GenServer' do
+   {status, _pid} = start_server()
+   assert status == :ok
+  end
+
+  test 'returns empty Map as a default state of GenServer' do
+    start_server()
+    assert CrashCourseAgent.get == %{}
+  end
+
+  test 'returns updated state after put action' do
+    start_server
+    CrashCourseAgent.put(:a, 1)
+    assert CrashCourseAgent.get() == %{a: 1}
+  end
+
+  test 'returns desired key value for get ' do
+    start_server()
+    set_state()
+
+    assert CrashCourseAgent.get(:a) == 1
+  end
+
+  test 'deletes desired key' do
+    start_server
+    set_state
```

Comment:
Put `set_state` inside `setup` clause as well, depending on context. To do that, first of all, declare the setup block with `context` param:
```elixir
setup context do
  IO.inspect(context, label: "CONTEXT")
end
```
Examine what the context is. Understand, how one might selectively run code based on what test we are currently setting up. Call `set_state/0` when needed from there.